packages/fx-core/src/core/middleware/envInfoLoader.ts

```diff
@@ -91,3 +115,76 @@ export async function loadSolutionContext(
 
   return ok(solutionContext);
 }
+
+async function askTargetEnvironment(
```

Comment:
Since we already have a [`QuestionModel middle`](https://github.com/OfficeDev/TeamsFx/blob/dev/packages/fx-core/src/core/middleware/questionModel.ts) to handle the user input for the lifecycle operations (such as provision, deploy, publish), can we also handle the "ask target environment" question, to keep single responsibility in the `envInfoLoader`?

Reply:
It's due to the logic in `QuestionModelMW` requires `solutionContext`, which needs to load env info before entering this middleware. In another word, `EnvInfoLoader` needs to be processed before `QuestionModelMW`, but `EnvInfoLoader` needs to ask user input before loading the env info.....

Reply:
I see, Thanks Qinen for the explanation.